File: Data Structures & Algorithms/products-of-array-discluding-self/submission-10.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res;
        int zero_count = count(nums.begin(), nums.end(), 0);
        int product = 1;
        for (int& num:nums) {
            if (num != 0) {
                product *= num;
            }
        }

        if (zero_count > 1) {
            vector<int> res(nums.size(), 0);
            return res;
        }

        if (zero_count == 1) {
            for (int& num:nums) {
                if (num != 0) {
                    res.push_back(0);
                } else {
                    res.push_back(product);
                }
            }

            return res;
        }

        for (int& num:nums) {
            int answer = product / num;
            res.push_back(answer);
            }

        return res;

    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-10.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        num_freq = {}

        for num in nums:
            num_freq[num] = num_freq.get(num, 0) + 1
        
        freq_num = []

        for num, freq in num_freq.items():
            freq_num.append([freq,num])

        freq_num.sort(reverse = True)

        res = []

        for freq, num in freq_num:
            if len(res) < k:
                res.append(num)

        return res